### ISSUE TITLE
Send mail not only via SMTP

### ISSUE THREAD

**Issue by stbc** [louislam/uptime-kuma#2793]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

It would be great if UK would not only send mail via smtp but optional via the mail subsystem on the system it is installed at.

### ✔️ Solution

Make use of system MTA to send mail to receiver

**Comment by apio-sys** [comment]
I don't think this needs any work since it is already working (for me at least). I use exim4 on a Debian 12 as MTA. Hence all apps on the server can simply dump a mail() command to get stuff out.
```
netstat -tulp | grep "exim"
tcp        0      0 localhost:smtp          0.0.0.0:*               LISTEN      1555/exim4          
tcp6       0      0 ip6-localhost:smtp      [::]:*                  LISTEN      1555/exim4
```
Provided that your MTA actually works, i.e. test with something like:

`echo "This is a test." | mail -s "Testing" mymail@domain.com`

you can setup Uptime Kuma alert with "SMTP" to your local MTA like so:

![Screenshot from 2024-02-01 14-08-31](https://github.com/louislam/uptime-kuma/assets/26541964/016c176f-1dd6-4459-8866-802ab0defc74)

and that's really all. To avoid confusion, maybe an entry could be added to say "Email (local)" without the unused fields from the SMTP setup to make this clearer. And/or in documentation. But the most important is that this actually works fine already so rest is nice2have/cosmetic and no showstopper.

**Comment by CommanderStorm** [comment]
Adding another notification type just with fewer fields is not something that, I think would be beneficial.

If you have ideas where helptexts/... would be benefitial, a Pull request would be appreciated

That our monitor/notification system is something that we are aware of, but at least I currently don't have the spare time to document this more than currently done in the frontend. Related issue https://github.com/louislam/uptime-kuma-wiki/issues/71

I am going to close this issue as reolved, thanks @apio-sys

**Comment by stbc** [comment]
Thx a lot for pointing that out, @apio-sys 
Never tried that somehow, as  @CommanderStorm said, some help would be nice :)

**Comment by apio-sys** [comment]
Would be glad helping document SMTP sending via local MTA. But just to make sure before issuing a PR, would it be wanted as inline help text inside the monitor here: https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/smtp.js or in the help document here: https://github.com/louislam/uptime-kuma-wiki/blob/master/Notification-Methods.md ?

Kind regards,

Joris.

**Comment by apio-sys** [comment]
OK thanks for the clarifications. I need to setup a local dev env to test some stuff then. Would something like this:
```

        <div class="mb-3">
            <label for="hostname" class="form-label">{{ $t("Hostname") }}</label>
            <input id="hostname" v-model="$parent.notification.smtpHost" type="text" class="form-control" required placeholder="This might be &quot;localhost&quot; if sending via a local MTA">
        </div>
```

help or not? In that case it won't be translated of course but I can't see where I could otherwise add help text. Maybe using a hover over a question mark (icon or not) and then get a help bubble that can then also be translated?

Sorry for the noise, this is really not a big issue since it does work fine already but might help some people that wonder to work with local MTA vs remote SMTP (as I did myself when I started testing this great software a few months back).

**Comment by CommanderStorm** [comment]
> Would something like this

Great question, actually we currently have two ways of doing this:
- the simple way is adding the following
  ```html
  <div class="form-text">{{ $t("Either enter the hostname of the server you want to contact or localhost if you intend to use a locally configured mail transfer agent") }}</div>
  ```
- using i18n-t components to embed more **complex translations** or **translations requiring html tags** (see discussion in #4135 why this is nessesary ^^)
  https://github.com/louislam/uptime-kuma/blob/56aa81e337043d4efa76c46dd7dc8320c9688398/src/components/notifications/Webhook.vue#L28-L33
  https://github.com/louislam/uptime-kuma/blob/56aa81e337043d4efa76c46dd7dc8320c9688398/src/components/notifications/Webhook.vue#L35-L37


I think for the usecase the second one fits better as likely a link to a piece of documentation/wikipedia what a MTA fits better

> Sorry for the noise

**This is NOT noise.**
Noise is something like `+1`-ing an issue, creating hard/impossible to reproduce issues, ...

This is valid (encouraged!) discussion and has the upsides (among others) of f.ex.
- knowing where other devs struggle => helping us improve the [contribution guide](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md) and
- not having to frustrate the heck out of you because merging the PR takes longer ^^

**Comment by apio-sys** [comment]
OK thanks. Once this is decided I would be glad to contribute to the Wiki article that would be referenced.

**Comment by CommanderStorm** [comment]
Once what is decided?

When I was saying that linking to docs/Wikipedia might be a good idea, I was thinking about the Windows/macOS/Linux docs[^1] about sending mails or the Wikipedia article explaining what an MTA is.
(=> i.e. the pointers you would find helpfull, if you were reading the UI for the first time ^^)
I don't think this needs a wiki article on our side (or does it??)

[^1]: have not looked if they exist

**Comment by apio-sys** [comment]
Well just to know we agree on to where to put that helptext and/or small wiki/doc. Either the simple way as per your point 1 and/or 2.a. or b. If we agree on where to put what, we can start working on it.

**Comment by CommanderStorm** [comment]
Yes, helptext below the hostname field is fine by me
